### PR TITLE
statistics for auto_shows and producers

### PR DESCRIPTION
**Pull request by VeraLapidus**

### CODE REVIEW

**Review by shprexen — CHANGES_REQUESTED** [review]
Put business logic into separate service, i.e. python file.
We should keep it in a separate service to test that functions in the future

File: auto_show/statistics.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    consumption = queryset_for_statistics.aggregate(Sum('price'))['price__sum']
+
+    # bought_cars
+    bought_cars = []
+    for deal in DealSerializer(queryset_for_statistics, many=True).data:
+        bought_cars.append(deal.get("car_instances"))
+
+    context = {'auto_show': auto_show.name, 'amount_of_bought_cars': amount_of_bought_cars,
+               'consumption, USD': consumption, 'bought_cars': bought_cars}
+    return Response(context)
```
Comment (shprexen) [inline auto_show/statistics.py]: return dict here. Response should be returned in the view

File: auto_show/statistics.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from django.db.models import Sum, Q
+from rest_framework.response import Response
+
+from auto_show.models import AutoShow
+from deals.models import Deal
+from deals.serializers import DealSerializer
+
+
+def statistics(request, pk=None):
```
Comment (shprexen) [inline auto_show/statistics.py]: why do you need request object here?

File: producer/statistics.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    # name and amount of unique clients
+    unique_clients_list = []
+    for deal in DealSerializer(queryset_for_statistics, many=True).data:
+        unique_clients_list.append(deal.get("auto_shows"))
+    unique_clients_set = set(unique_clients_list)
+    amount_of_unique_clients = len(unique_clients_set)
+
+    context = {'producer_name': producer.name, 'amount_of_sold_cars': amount_of_sold_cars, 'profit': profit,
+               'unique_clients': unique_clients_set, 'amount_of_unique_clients': amount_of_unique_clients}
+    return Response(context)
```
Comment (shprexen) [inline producer/statistics.py]: the same as mentioned above

**Review by shprexen — CHANGES_REQUESTED** [review]

File: auto_show/statistics.py (hunk 1 above)
Comment (VeraLapidus) [reply auto_show/statistics.py]: done

File: auto_show/statistics.py (hunk 2 above)
Comment (VeraLapidus) [reply auto_show/statistics.py]: done

File: producer/statistics.py (hunk 3 above)
Comment (VeraLapidus) [reply producer/statistics.py]: done

File: auto_show/statistics.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+from django.db.models import Sum, Q
+
+from auto_show.models import AutoShow
+from deals.models import Deal
+from deals.serializers import DealSerializer
+
+
+def statistics(pk):
+    auto_show = AutoShow.objects.get(pk=pk)
+    queryset_for_statistics = Deal.objects.filter(Q(participants="Producer-AutoShow") & Q(auto_shows=auto_show.id))
```
Comment (shprexen) [inline auto_show/statistics.py]: We shouldn't use magic strings, i.e. `Producer-AutoShow`
Replace it with the value of `Participants. PRODUCER_AUTOSHOW`

File: producer/statistics.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+from django.db.models import Sum, Q
+
+from deals.models import Deal
+from deals.serializers import DealSerializer
+from producer.models import Producer
+
+
+def statistics(pk):
+    producer = Producer.objects.get(pk=pk)
+    queryset_for_statistics = Deal.objects.filter(Q(participants="Producer-AutoShow") & Q(producers=producer.id))
```
Comment (shprexen) [inline producer/statistics.py]: The same as mentioned above

**Review by shprexen — CHANGES_REQUESTED** [review]

File: auto_show/statistics.py (hunk 4 above)
Comment (VeraLapidus) [reply auto_show/statistics.py]: done

File: producer/statistics.py (hunk 5 above)
Comment (VeraLapidus) [reply producer/statistics.py]: done

**Review by shprexen — APPROVED** [review]

**Pull request merged**